# Ejercicios de Distribución Exponencial y de Poisson #

# Distribución Exponencial

<span style="color: blue;">Economía</span>.  El Mar de Cortés, en México, es la zona donde se localiza el mayor número de granjas dedicadas al cultivo de ostras perleras. Según los productores, 3.6 de cada cien ostras pueden llegar a tener un buen valor comercial. Si tenemos 150 ostras, 

a) Determina la probabilidad de que 12 de ellas tengan perlas de buen valor comercial. 

b) Encuentra la probabilidad de que cuando mucho dos tengan perlas de buen valor comercial.

Sea:

$$X=\mbox{ Número de ostras (150 ostras)}$$

Como se analiza una muestra de 150 ostras y cada una tiene una probabilidad  p = 0.036  de tener una perla valiosa:
$$p = \frac{3.6}{100} = 0.036$$
$$X∼\mbox{Binomial}(n=150,p=0.036)$$

## a) Determina la probabilidad de que 12 de ellas tengan perlas de buen valor comercial. 
$$
\begin{align*}
P(X=12) &= \binom{150}{12}(0.036)^{12}(0.964)^{150-12} \\
       &= \binom{150}{12}(0.036)^ {12}(0.964)^{138} \\
       &= 172420656389440550(0.000000000000000004738381338321612)(0.006347908799182596) \\
       &=0.00518593
\end{align*}
$$

Por lo tanto:

$$P(X=12)\approx 0.005186 $$

Es decir, aproximadamente:


$$ 0.5186 \%$$

## b) Encuentra la probabilidad de que cuando mucho dos tengan perlas de buen valor comercial.


“Al menos dos” significa:

$$P(X \ge 2)$$

Conviene usar el complemento:

$$P(X \ge 2) = 1 − P(X < 2)$$

Es decir:

$$P(X \ge 2)=1−[P(X=0)+P(X=1)]$$

Por lo tanto: 
$$
\begin{align*}
P(X \geq 2) &= 1 - P(X < 2) \\
            &= 1 - [P(X=0) + P(X=1)] \\
            &= 1 - \left[\binom{150}{0}(0.036)^0(0.964)^{150}
            + \binom{150}{1}(0.0360)^1(0.964)^149\right] \\
            &= 1 - \left[(1)(1)(0.964)^{150}
            + 150(0.036)(0.964)^{149}\right] \\
            &= 1 - [0.004088 + 0.022902] \\
            &= 1 - 0.02699 \\
            &= 0.97301
\end{align*}
$$

Por lo tanto:

$$P(X \ge 2) \approx 0.97301$$

Es decir, aproximadamente: 
$$97.301\%$$.


In [3]:
import matplotlib.pyplot as plt
from scipy.stats import binom

# Parámetros
n = 150
p = 0.036

# Valores posibles de X
x = list(range(n + 1))

# Distribución binomial
probabilidades = binom.pmf(x, n, p)

# Mostrar tabla en consola
#for k, prob in zip(x, probabilidades):
#    print(f"P(X = {k}) = {prob:.6f}")

# ==========================
# Generar tabla HTML
# ==========================

html = """
<table style="border-collapse: collapse; width: 420px; text-align: center;">
  <thead>
    <tr>
      <th style="border: 1px solid #000; padding: 10px; width: 160px;">x</th>
      <th style="border: 1px solid #000; padding: 10px; width: 260px;">P(X=x)</th>
    </tr>
  </thead>
  <tbody>"""

for k, prob in zip(x, probabilidades):
    html += f"""
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">{k}</td>
      <td style="border: 1px solid #000; padding: 8px;">{prob:.6f}</td>
    </tr>"""

html += """
  </tbody>
</table>
"""

# Mostrar el HTML en la consola
#print("\nCódigo HTML:\n")
#print(html)




<table style="border-collapse: collapse; width: 420px; text-align: center;">
  <thead>
    <tr>
      <th style="border: 1px solid #000; padding: 10px; width: 160px;">x</th>
      <th style="border: 1px solid #000; padding: 10px; width: 260px;">P(X=x)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">0</td>
      <td style="border: 1px solid #000; padding: 8px;">0.004088</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">1</td>
      <td style="border: 1px solid #000; padding: 8px;">0.022902</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">2</td>
      <td style="border: 1px solid #000; padding: 8px;">0.063717</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">3</td>
      <td style="border: 1px solid #000; padding: 8px;">0.117387</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">4</td>
      <td style="border: 1px solid #000; padding: 8px;">0.161103</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">5</td>
      <td style="border: 1px solid #000; padding: 8px;">0.175675</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">6</td>
      <td style="border: 1px solid #000; padding: 8px;">0.158545</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">7</td>
      <td style="border: 1px solid #000; padding: 8px;">0.121799</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">8</td>
      <td style="border: 1px solid #000; padding: 8px;">0.081304</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">9</td>
      <td style="border: 1px solid #000; padding: 8px;">0.047906</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">10</td>
      <td style="border: 1px solid #000; padding: 8px;">0.025225</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">11</td>
      <td style="border: 1px solid #000; padding: 8px;">0.011989</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">12</td>
      <td style="border: 1px solid #000; padding: 8px;">0.005186</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">13</td>
      <td style="border: 1px solid #000; padding: 8px;">0.002056</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">14</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000751</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">15</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000254</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">16</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000080</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">17</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000024</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">18</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000007</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">19</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000002</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">20</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">21</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">22</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">23</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">24</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">25</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">26</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">27</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">28</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">29</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">30</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">31</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">32</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">33</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">34</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">35</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">36</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">37</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">38</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">39</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">40</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">41</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">42</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">43</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">44</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">45</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">46</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">47</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">48</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">49</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">50</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">51</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">52</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">53</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">54</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">55</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">56</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">57</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">58</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">59</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">60</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">61</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">62</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">63</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">64</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">65</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">66</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">67</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">68</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">69</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">70</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">71</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">72</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">73</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">74</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">75</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">76</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">77</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">78</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">79</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">80</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">81</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">82</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">83</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">84</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">85</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">86</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">87</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">88</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">89</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">90</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">91</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">92</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">93</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">94</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">95</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">96</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">97</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">98</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">99</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">100</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">101</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">102</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">103</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">104</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">105</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">106</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">107</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">108</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">109</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">110</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">111</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">112</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">113</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">114</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">115</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">116</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">117</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">118</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">119</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">120</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">121</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">122</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">123</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">124</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">125</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">126</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">127</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">128</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">129</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">130</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">131</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">132</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">133</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">134</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">135</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">136</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">137</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">138</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">139</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">140</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">141</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">142</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">143</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">144</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">145</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">146</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">147</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">148</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">149</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px;">150</td>
      <td style="border: 1px solid #000; padding: 8px;">0.000000</td>
    </tr>
  </tbody>
</table>

# Distribución de Poisson


<span style="color: blue;">Deportes.</span>En un gimnasio sólo para mujeres se realizan ejercicios aeróbicos y ejercicios en
aparatos. El circuito está constituido por nueve aparatos tonificadores y ocho estaciones para
los ejercicios aeróbicos. En un día cualquiera, en hora pico (entre las 19 y 20 horas), llegan enpromedio siete mujeres. En ese horario:


a) ¿Cuál es la probabilidad de que lleguen menos de cinco mujeres?.

b) ¿Cuál es la probabilidad de que estén ocupados todos los aparatos?

c) Y en un día extraordinario, ¿cuál es la probabilidad de que tanto los aparatos como las estaciones estén ocupados?

Sea:

$$X=\text{ número de mujeres que llegan al gimnasio en hora pico}$$

Como se conoce el número promedio de mujeres que llegan al gimnasio en hora pico y se desea modelar número de personas que asisten al establecimiento en ese horario, utilizamos una **distribución de Poisson** con parámetro:


$$\lambda=7$$

Por lo tanto:

$$X\sim\operatorname{Poisson}(7)$$

La función de probabilidad de Poisson es:


$$P(X=x)=\frac{e^{-\lambda}\lambda^x}{x!}, \quad \mbox{con} \quad x=0,1,2,\ldots$$

Para este problema:


$$P(X=x)=\frac{e^{-7}(7)^x}{x!}$$

In [6]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson
from IPython.display import HTML, display

# Parámetro de la distribución
lam = 7

# Valores que se mostrarán en la tabla y en la gráfica
x = np.arange(0, 17)

# Probabilidades P(X = x)
probabilidades = poisson.pmf(x, mu=lam)

# Construcción de la tabla HTML
html = """
<table style="
    border-collapse: collapse;
    width: 520px;
    text-align: center;
    font-family: Arial, sans-serif;
">
    <thead>
        <tr>
            <th style="
                border: 1px solid #000;
                padding: 10px;
                width: 200px;
            ">x</th>
            <th style="
                border: 1px solid #000;
                padding: 10px;
                width: 320px;
            ">P(X = x)</th>
        </tr>
    </thead>
    <tbody>
"""

for valor, probabilidad in zip(x, probabilidades):
    html += f"""
        <tr>
            <td style="border: 1px solid #000; padding: 8px;">
                {valor}
            </td>
            <td style="border: 1px solid #000; padding: 8px;">
                {probabilidad:.6f}
            </td>
        </tr>
    """

html += """
    </tbody>
</table>
"""

display(HTML(html))

x,P(X = x)
0,0.000912
1,0.006383
2,0.022341
3,0.052129
4,0.091226
5,0.127717
6,0.149003
7,0.149003
8,0.130377
9,0.101405


## a) ¿Cuál es la probabilidad de que lleguen menos de cinco mujeres?.

Se desea calcular:


$$P(X<5)$$

Esto equivale a sumar las probabilidades individuales desde cero hasta cuatro mujeres:



$$P(X < 5) = P(X = 0) + P(X = 1) + P(X = 2) + P(X = 3) + P(X = 4)$$



Sustituimos los valores correspondientes de la tabla donde ya sustituimos el promedio ($\lambda = 7$) para cada valor de $x$ en la función de probabilidad de Poisson:

$P(X = 0) = 0.000912$

$P(X = 1) = 0.006383$

$P(X = 2) = 0.022341$

$P(X = 3) = 0.052129$

$P(X = 4) = 0.091226$


Realizamos la suma de las probabilidades obtenidas:


$$P(X < 5) = 0.000912 + 0.006383 + 0.022341 + 0.052129 + 0.091226$$

Por lo tanto:
$$P(X < 5) = 0.172991$$

La probabilidad de que lleguen menos de cinco mujeres es aproximadamente 17.2991%.





## b) Probabilidad de que estén ocupados todos los aparatos
Se desea calcular:
$$P(X \ge 9)$$
Para calcularlo de forma más sencilla, utilizamos la propiedad del evento complementario:

$$P(X \ge 9) = 1 - P(X \le 8)$$Donde $P(X \le 8)$


es la suma de las probabilidades desde cero hasta ocho mujeres:



$$P(X \le 8) = P(X = 0) + P(X = 1) + P(X = 2) + P(X = 3) + P(X = 4) + P(X = 5) + P(X = 6) + P(X = 7) + P(X = 8)$$



Sustituimos los valores correspondientes (utilizando los datos de la tabla y completando $x=7$ y $x=8$ ):

$P(X = 0) = 0.000912$

$P(X = 1) = 0.006383$

$P(X = 2) = 0.022341$

$P(X = 3) = 0.052129$

$P(X = 4) = 0.091226$

$P(X = 5) = 0.127717$

$P(X = 6) = 0.149003$

$P(X = 7) = 0.149003$

$P(X = 8) = 0.130377$

Realizamos la suma acumulada hasta ocho:

$$P(X \le 8) = 0.000912 + 0.006383 + 0.022341 + 0.052129 + 0.091226 + 0.127717 + 0.149003 + 0.149003 + 0.130377$$$$P(X \le 8) = 0.729091$$

Finalmente, restamos este resultado de la unidad para obtener el complemento:

$$P(X \ge 9) = 1 - 0.729091$$Por lo tanto:$$P(X \ge 9) = 0.270909$$

La probabilidad de que estén ocupados todos los aparatos es aproximadamente 27.09%.

## c) Probabilidad de que tanto los aparatos como las estaciones estén ocupados

Se desea calcular:
$$P(X \ge 17)$$

Utilizamos la propiedad del evento complementario para simplificar el cálculo:

$$P(X \ge 17) = 1 - P(X \le 16)$$Donde $P(X \le 16)$

es la suma de las probabilidades individuales desde cero hasta dieciséis mujeres:


$$P(X \le 16) = P(X = 0) + P(X = 1) + \dots + P(X = 16)$$

Sustituimos los valores correspondientes para una distribución de Poisson con promedio $\lambda = 7$ :
* **$P(X = 0) = 0.000912$**
* **$P(X = 1) = 0.006383$**
* **$P(X = 2) = 0.022341$**
* **$P(X = 3) = 0.052129$**
* **$P(X = 4) = 0.091226$**
* **$P(X = 5) = 0.127717$**
* **$P(X = 6) = 0.149003$**
* **$P(X = 7) = 0.149003$**
* **$P(X = 8) = 0.130377$**
* **$P(X = 9) = 0.101405$**
* **$P(X = 10) = 0.070983$**
* **$P(X = 11) = 0.045171$**
* **$P(X = 12) = 0.026350$**
* **$P(X = 13) = 0.014188$**
* **$P(X = 14) = 0.007094$**
* **$P(X = 15) = 0.003311$**
* **$P(X = 16) = 0.001448$**



Realizamos la suma acumulada hasta dieciséis:

$$P(X \le 16) = 0.000912 + 0.006383 + \dots + 0.001448$$$$P(X \le 16) = 0.999341$$

Finalmente, restamos este resultado de la unidad para obtener el complemento:

$$P(X \ge 17) = 1 - 0.999341$$Por lo tanto:$$P(X \ge 17) = 0.000659$$

La probabilidad de que tanto los aparatos como las estaciones estén ocupados es aproximadamente 0.07%.